### PR TITLE
test with and without local cache

### PR DESCRIPTION
**Pull request by grosser**
@mperham @petergoldstein

a preview of what I have in mind to tighten local cache testing, looks good ? -> I'll convert more tests and kick out the standalone LocaleCache tests

decided against using something that is inside the testcase so that you can dig into a specific issue with `ruby xxx_test.rb -n 'bla bla with LocaleCache'`

### CODE REVIEW

File: test/test_active_support.rb [hunk 1]
```diff
@@ -9,6 +9,23 @@ def cache_key
...
+    it "#{message} with LocaleCache" do
+      with_cache do
+        @dalli.with_local_cache do
+          instance_eval(&block)
+        end
+      end
+    end
+
+    it "#{message} without LocaleCache" do
+      with_cache do
```
Comment (petergoldstein) [inline test/test_active_support.rb]: Looks like there's a typo in these two messages - 'LocaleCache'.  Plus I'm not really nuts about the method name - very obscure.  Can you come up with something clearer, although potentially more verbose?

Comment (grosser) [inline test/test_active_support.rb]: `it_with_and_without_locale_cache` ?

Comment (petergoldstein) [inline test/test_active_support.rb]: `it_with_and_without_local_cache` (same typo) would certainly be clearer.

Comment (grosser) [inline test/test_active_support.rb]: dammit ... I do this typo way too often ... working to much in i18n ;D

**Comment by grosser** [comment]
updated all that made sense and did not fail

**Comment by grosser** [comment]
best viewed [without whitespace](https://github.com/mperham/dalli/pull/570/files?w=1)

**Comment by grosser** [comment]
@petergoldstein

**Comment by grosser** [comment]
@petergoldstein does this look good ?

**Comment by grosser** [comment]
@mperham @petergoldstein can you please review this ?

**Comment by petergoldstein** [comment]
@grosser Conceptually this seems fine.  If you'd like to work it into a full PR I'd be happy to take a look.  Thanks!

**Comment by grosser** [comment]
I think it's done now, all tests converted that are make sense to run with and without.

**Comment by petergoldstein** [comment]
Thanks!

**Pull request merged**